runtime/oti/j9nonbuilder.h

```diff
@@ -262,6 +262,12 @@ typedef enum {
 	J9FlushCompQueueDataBreakpoint
 } J9JITFlushCompilationQueueReason;
 
+typedef enum J9SharedClassCacheMode {
```

Comment:
We have 
`#if defined(J9VM_OPT_SHARED_CLASSES)`
in this file. 

Can you move `J9SharedClassCacheMode` inside the `#if defined(J9VM_OPT_SHARED_CLASSES)`